_posts/2021-02-18-survey/survey.Rmd

```diff
@@ -0,0 +1,314 @@
+
+And now, on to Spark.
+
+# Spark
+
+### Areas and applications
+
+For Spark, questions broadly paralleled those asked for DL.
+
+Overall, judging from this survey (and unsurprisingly), Spark is predominantly used in industry (n = 39). For academic staff and students, n = 4 (each). Seventeen people reported using Spark in their spare time, while 34 said they wanted to use it in the future.
+
+Looking at industry sectors, we again find finance, consulting, and healthcare dominating.[^5]
+
+[^5]: In hindsight, if you were a consultant, the survey should have asked what industry sectors you were doing consulting for/in.
+
+```{r, eval = TRUE, echo = FALSE, layout="l-body-outset", fig.cap="Number of users reporting to use Spark in industry. Smaller groups not displayed."}
+knitr::include_graphics("images/spark_ind.png")
+
+```
+
+What do survey respondents do with Spark? Analyses of tabular data and time series dominate:
+
+```{r, eval = TRUE, echo = FALSE, layout="l-body-outset", fig.cap="Number of users reporting to use Spark in industry. Smaller groups not displayed."}
+knitr::include_graphics("images/spark_appl.png")
+
+```
+
+### Frameworks and skills
+
+As with deep learning, we wanted to know what language people use to do Spark. If you look at the below graphic, you see R appearing twice: once in connection with `sparklyr`, once with `SparkR`. What's that about?
+
+Both `sparklyr` and `SparkR` are R interfaces for Apache Spark, each designed and built with very different sets of priorities and, consequently, trade-offs in mind.
+
+[`sparklyr`](https://spark.rstudio.com/), one the one hand, will appeal to data scientists at home in the tidyverse, as they'll be able to use all the data manipulation interfaces they're familiar with from packages such as `dplyr`, `DBI`, `tidyr`, or `broom`.
+
+[`SparkR`](https://spark.apache.org/docs/latest/sparkr.html), on the other hand, is a light-weight R binding for Apache Spark, and bundled with the same. It's an excellent choice for practitioners who are well-versed in Apache Spark and just need a thin wrapper to access various Spark functionalities from R.
+
+```{r, eval = TRUE, echo = FALSE, layout="l-body-outset", fig.cap="Language / language bindings used to do Spark."}
+knitr::include_graphics("images/spark_lang.png")
+
+```
+
+When asked to rate their expertise in R[^6] and Spark, respectively, respondents showed similar behavior as observed for deep learning above: Most people seem to think more of their R skills than their theoretical Spark-related knowledge. However, the same caution should be exercised here as above, -- or more caution, even: The number of responses here amounted to about half of its DL counterpart.
```

Comment:
```suggestion
When asked to rate their expertise in R[^6] and Spark, respectively, respondents showed similar behavior as observed for deep learning above: Most people seem to think more of their R skills than their theoretical Spark-related knowledge. However, even more caution should be exercised here as above: The number of responses here amounted to about half of its DL counterpart.
```

Reply:
> However, even more caution should be exercised here as above:

that's much nicer than the clumsy thing I wrote

however, "as above"? shouldn't it be "than above"?

Reply:
ended up with
>However, even more caution should be exercised here than above: The number of responses here was significantly lower.